src/controller/category.controller.ts

```diff
@@ -0,0 +1,18 @@
+import { NextFunction, Request, Response } from 'express';
+import { nanoid } from 'nanoid';
+import AppError from '../class/AppError';
+import CategoryService from '../services/category.service';
+
+export const createCategory = async (
+  req: Request,
+  res: Response,
+  next: NextFunction
+) => {
+  try {
+    const { name } = req.body;
+    const category = await CategoryService.create(name);
```

Comment:
Asyncronously wait for the servie to return a valid data before sending back a response.